db/migrate/20160620205641_add_property_attributes_to_generic_object.rb

```diff
@@ -0,0 +1,46 @@
+class AddPropertyAttributesToGenericObject < ActiveRecord::Migration[5.0]
+  class CustomAttribute < ActiveRecord::Base; end
+
+  class GenericObject < ActiveRecord::Base
+    belongs_to :generic_object_definition, :class_name => 'AddPropertyAttributesToGenericObject::GenericObjectDefinition'
+  end
+
+  class GenericObjectDefinition < ApplicationRecord
+    serialize :properties, Hash
+    has_many  :generic_objects, :class_name => 'AddPropertyAttributesToGenericObject::GenericObject'
+
+    TYPE_MAP = {
+      :boolean  => ActiveRecord::Type::Boolean.new,
+      :datetime => ActiveRecord::Type::DateTime.new,
+      :time     => ActiveRecord::Type::Time.new,
+      :float    => ActiveRecord::Type::Float.new,
+      :integer  => ActiveRecord::Type::Integer.new,
+      :string   => ActiveRecord::Type::String.new,
+    }.freeze
+
+    def type_cast(attr_name, value)
+      TYPE_MAP.fetch(properties[:attributes][attr_name]).cast(value)
+    end
+  end
+
+  def up
+    add_column :generic_objects, :property_attributes, :jsonb, :default => {}
+
+    say_with_time("Migrating custom attributes into generic object's properties column") do
```

Comment:
@lfu @bzwei Since this feature was never fully implemented is there any need to migrate this data?